### PR TITLE
MinuteMedia: Add new bidder

### PR DESCRIPTION
**Pull request by Net-burst**

### CODE REVIEW

File: src/test/java/org/prebid/server/it/MinuteMediaTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import static java.util.Collections.singletonList;
+
+@RunWith(SpringRunner.class)
+public class MinuteMediaTest extends IntegrationTest {
+
+    @Test
+    public void openrtb2AuctionShouldRespondWithBidsFromMinuteMedia() throws IOException, JSONException {
+        // given
+
+        WIRE_MOCK_RULE.stubFor(post(urlPathEqualTo("/minutemedia-exchange"))
```
Comment (SerhiiNahornyi) [inline src/test/java/org/prebid/server/it/MinuteMediaTest.java:25]: As you added query param to test application props pls add `withQueryParam` check

File: src/main/java/org/prebid/server/bidder/minutemedia/model/ExtImpMinuteMedia.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.prebid.server.bidder.minutemedia.model;
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/minutemedia/model/ExtImpMinuteMedia.java:1]: Should be part of proto package

File: src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        for (Imp imp : imps) {
+            if (imp.getId().equals(bid.getImpid())) {
+                if (imp.getBanner() != null) {
+                    return BidType.banner;
+                } else if (imp.getVideo() != null) {
+                    return BidType.video;
+                } else if (imp.getXNative() != null) {
+                    return BidType.xNative;
+                } else if (imp.getAudio() != null) {
+                    return BidType.audio;
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java:119]: Pls cover native and banner with unit tests

File: src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid, getBidType(bid, bidRequest.getImp()), bidResponse.getCur()))
+                .toList();
+    }
+
+    private static BidType getBidType(Bid bid, List<Imp> imps) {
+        for (Imp imp : imps) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java:110]: From go they use mtype check

File: src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final String orgId;
+
+        try {
+            orgId = extractFirstImpOrdId(bidRequest.getImp());
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java:51]: They are checking each imp for org, not just first

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java (hunk 5 above)
Comment (Net-burst) [reply src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java:51]: Only first. They iterate over `request.imp` object, but they return right away.
```go
func extractOrg(openRTBRequest *openrtb2.BidRequest) (string, error) {
	var err error
	for _, imp := range openRTBRequest.Imp {
		var bidderExt adapters.ExtImpBidder
		if err = json.Unmarshal(imp.Ext, &bidderExt); err != nil {
			return "", fmt.Errorf("failed to unmarshal bidderExt: %w", err)
		}

		var impExt openrtb_ext.ImpExtMinuteMedia
		if err = json.Unmarshal(bidderExt.Bidder, &impExt); err != nil {
			return "", fmt.Errorf("failed to unmarshal ImpExtMinuteMedia: %w", err)
		}

		return strings.TrimSpace(impExt.Org), nil
	}

	return "", errors.New("no imps in bid request")
}
```

Comment (Net-burst) [reply src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java:51]: I can't say that I like how it's done, but there was a discussion about it and they made an explicit decision to not do any kind of validation for this field...

Comment (SerhiiNahornyi) [reply src/main/java/org/prebid/server/bidder/minutemedia/MinuteMediaBidder.java:52]: Agree, `org` is the required field and always will be here

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**